src/backend/gl/src/queue.rs

```diff
@@ -835,7 +895,26 @@ impl hal::queue::RawCommandQueue<Backend> for CommandQueue {
         S: 'a + Borrow<native::Semaphore>,
         Iw: IntoIterator<Item = &'a S>,
     {
+        let gl = &self.share.context;
+
         for swapchain in swapchains {
+            let extent = swapchain.0.borrow().extent;
+
+            gl.BlitNamedFramebuffer(
+                self.state.fbo,              // Default framebuffer 
```

Comment:
this shouldn't be the current fbo but rather a swapchain owned fbo, which has the backbuffer image attached.

Reply:
When we do a renderpass we bind a framebuffer, and we want to draw to that framebuffer, right?

Reply:
yes, but a framebuffer is only a container object. At the end we want to display the content of the backbuffer images. So we want to blit the color images of the backbuffer to the display framebuffer. The currently bound fbo might not include the current backbuffer image we want to present.

Reply:
The swapchain doesn't own framebuffers in any other backend, I'm confused why it would own one in the gl backend? The framebuffer is just something to write to, right? So we have multiple framebuffers for our backbuffer images so we can write to them while another one is displaying. In this case when we do a renderpass we bind a framebuffer, which changes the state fbo, and then we draw to that fbo, and then blit from that fbo to default... I'm not really sure what the change would be here

Reply:
> The swapchain doesn't own framebuffers in any other backend, I'm confused why it would own one in the gl backend?

To be sure OpenGL Framebuffers and Vulkan Framebuffers are not exactly the same concept. At higher level, they are equal in terms of being a collection of attachments which will be used in draw commands, but differ due to the concept of renderpasses introduced by vulkan, which lead to a slight separation of responsibilites: a framebuffer alone only stores attachment while a subpasses hold the additional information about which attachments will be used during the draw commands.

> The framebuffer is just something to write to, right?

Just but we don't write to the framebuffer but rather access the attachments associated with it.

> So we have multiple framebuffers for our backbuffer images so we can write to them while another one is displaying

The backbuffer framebuffers would be purely for blitting and we don't bind them for any draw operations.

> In this case when we do a renderpass we bind a framebuffer, which changes the state fbo, and then we draw to that fbo, and then blit from that fbo to default... I'm not really sure what the change would be here

An example which should better show the intended logic:
```rust
cmd_buffer.begin_renderpass(pass1);  // bind GL-FBO of first subpass (e.g contains 2 color targets (one of these is present image #2) and one depth stencil)
..
cmd_buffer.next_subpass(); // bind GL-FBO of second subpass (e.g containts 1 color target (non-present image) and depth-stencil
...
cmd_buffer.end_renderpass();
...

queue.submit(cmd_buffer);
queue.present(); // currently bound GL-FBO of second subpass, we want to present image #1 of prev frame -> we bind a GL-FBO which only contains present image #1 and blit it to our display FBO
```

Reply:
That would complicate the backend and make it slow imo. If we have a framebuffer per subpass we'd have to blit each framebuffer into the default framebuffer. Rather than just keeping a single framebuffer for an entire renderpass.